src/schemathesis/models.py

```diff
@@ -441,6 +462,18 @@ def full_path(self) -> str:
     def links(self) -> Dict[str, Dict[str, Any]]:
         return self.schema.get_links(self)
 
+    def add_parameter(self, parameter: P) -> None:
+        lookup_table = {
+            "path": self.path_parameters,
+            "header": self.headers,
+            "cookie": self.cookies,
+            "query": self.query,
+            "body": self.body,
+        }
+        if parameter.location in lookup_table:
```

Comment:
Not sure if this condition is needed. If the parameter has a wrong location, then it should fail fast. May be `assert` will be enough or some internal exception